bsp_diff/common/kernel/linux-intel-lts2021/0028-Add-88q2122-eth-phy-suspend-resume-code.patch

```diff
@@ -0,0 +1,85 @@
+From 3f84f4ddf83036d5b839f2e859f63bd183861c9d Mon Sep 17 00:00:00 2001
+From: Zhao Ye <zhao.ye@intel.com>
+Date: Fri, 3 Nov 2023 10:58:47 +0800
+Subject: [PATCH] Add 88Q2122 eth phy suspend resume code
+
+Add 88Q2122 eth phy suspend resume code,
+Delete some useless warning.
+
+Tests:
+on A1 board, ethernet can suspend/resume
+success, ethernet power go down when it suspend.
+
+Tracked-On: OAM-113125
+Signed-off-by: Zhao Ye <zhao.ye@intel.com>
+---
+ drivers/net/phy/marvell-88q2xxx.c | 17 +++++++++++++++--
+ drivers/net/phy/phy_device.c      |  5 +++--
+ 2 files changed, 18 insertions(+), 4 deletions(-)
+
+diff --git a/drivers/net/phy/marvell-88q2xxx.c b/drivers/net/phy/marvell-88q2xxx.c
+index c853b72ec856..c2c70a5ffa10 100644
+--- a/drivers/net/phy/marvell-88q2xxx.c
++++ b/drivers/net/phy/marvell-88q2xxx.c
+@@ -52,18 +52,28 @@ static int mv88q2xxx_tx_enable(struct phy_device *phydev)
+ 
+ static int mv88q2xxx_tx_disable(struct phy_device *phydev)
+ {
+-	return phy_clear_bits_mmd(phydev, 3 ,0x8000, 0x8);
++	return phy_set_bits_mmd(phydev, 3 ,0x8000, 0x8);
+ }
+ 
+ static int mv88q2xxx_resume(struct phy_device *phydev)
+ {
++	phy_clear_bits_mmd(phydev, 3 ,0x801c, 0x1);
++	phy_clear_bits_mmd(phydev, 1 ,0x0, 0x800);
++	phy_clear_bits_mmd(phydev, 1 ,0x900, 0x800);
++	phy_clear_bits_mmd(phydev, 3 ,0x0, 0x800);
++
+ 	return mv88q2xxx_tx_enable(phydev);
+ }
+ 
+ static int mv88q2xxx_suspend(struct phy_device *phydev)
+ {
+-	return mv88q2xxx_tx_disable(phydev);
++	mv88q2xxx_tx_disable(phydev);
```

Comment:
should we handle disable failure case or return something?

Reply:
updated.